File: ios/App/App/StatusBarHeightPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN macro.
CAP_PLUGIN(StatusBarHeightPlugin, "StatusBarHeight",
           CAP_PLUGIN_METHOD(getHeight, CAPPluginReturnPromise);
) 

File: ios/App/App/StatusBarHeightPlugin.swift
import Foundation
import Capacitor
import UIKit

@objc(StatusBarHeightPlugin)
public class StatusBarHeightPlugin: CAPPlugin {
    @objc func getHeight(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            let height = UIApplication.shared.statusBarFrame.size.height
            call.resolve(["height": height])
        }
    }
} 